File: app/src/main/java/com/galtashma/parsedashboard/adapters/ParseClassesAdapter.java
package com.galtashma.parsedashboard.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import androidx.annotation.NonNull;

import com.galtashma.parsedashboard.R;
import com.lucasurbas.listitemview.ListItemView;
import com.parse.ParseSchema;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;


public class ParseClassesAdapter extends ArrayAdapter<ParseSchema> {

    public interface OnClickListener {
        void onSchemaClicked(ParseSchema schema);
    }

    private OnClickListener listener;

    public ParseClassesAdapter(@NonNull Context context) {
        super(context, R.layout.list_item);
    }

    public ParseClassesAdapter(@NonNull Context context, @NonNull List<ParseSchema> objects) {
        super(context, R.layout.list_item, objects);
    }

    @NonNull
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        final ParseSchema schema = getItem(position);
        if (convertView == null) {
            convertView = LayoutInflater.from(this.getContext()).inflate(R.layout.list_item, parent, false);
        }

        ListItemView item = (ListItemView) convertView;

        item.setTitle(schema.getName());
        item.setSubtitle(getItemCountString(schema) + getSchemaString(schema));

        item.setOnClickListener(view -> {
            if (listener != null){
                listener.onSchemaClicked(schema);
            }
        });

        return convertView;
    }

    private String getItemCountString(ParseSchema schema) {
        int itemCount = schema.getCountIfFetched();
        if (itemCount != -1) {
            return "("+itemCount+")\t";
        }

        schema.setOnCountListener((count, exception) -> notifyDataSetChanged());

        return "";
    }

    private String getSchemaString(ParseSchema schema){
        List<String> list = new ArrayList<>(schema.getFields().keySet());
        Collections.sort(list);
        return list.toString().replaceAll("\\[|\\]","").replaceAll(","," ");
    }

    public void setListener(OnClickListener listener) {
        this.listener = listener;
    }
}

File: app/src/main/java/com/galtashma/parsedashboard/adapters/ParseClassesAdapter.kt
package com.galtashma.parsedashboard.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.annotation.NonNull
import com.galtashma.parsedashboard.R
import com.lucasurbas.listitemview.ListItemView
import com.parse.ParseSchema

/**
 * Created by gal on 3/16/18, rewritten by Cyb3rKo on 05/12/22.
 */

class ParseClassesAdapter(
    @NonNull context: Context
) : ArrayAdapter<ParseSchema>(context, R.layout.list_item) {

    interface OnClickListener {
        fun onSchemaCLicked(schema: ParseSchema)
    }

    private lateinit var localListener: OnClickListener

    override fun getView(
        position: Int,
        convertView: View?,
        parent: ViewGroup
    ): View {
        var finalView = convertView
        val schema = getItem(position)
        if (convertView == null) {
            finalView = LayoutInflater.from(context).inflate(R.layout.list_item, parent, false)
        }

        val item = finalView as ListItemView

        item.title = schema?.name
        if (schema != null) {
            item.subtitle = getItemCountString(schema) + getSchemaString(schema)
        }

        item.setOnClickListener {
            if (schema != null) {
                localListener.onSchemaCLicked(schema)
            }
        }

        return finalView
    }

    private fun getItemCountString(schema: ParseSchema): String {
        val itemCount = schema.countIfFetched
        if (itemCount != -1) {
            return "($itemCount)\t"
        }

        schema.setOnCountListener { _, _ -> notifyDataSetChanged() }

        return ""
    }

    private fun getSchemaString(schema: ParseSchema): String {
        val list = schema.fields.keys.toList().sorted()
        return list.toString().replace("\\[|\\]", "").replace(",", " ")
    }

    fun setListener(listener: OnClickListener) {
        this.localListener = listener
    }
}
